src/Security/Authentication/Certificate/src/CertificateValidationCache.cs

```diff
@@ -43,8 +44,12 @@ public CertificateValidationCache(IOptions<CertificateValidationCacheOptions> op
         /// <param name="certificate">The certificate.</param>
         /// <param name="result">the <see cref="AuthenticateResult"/></param>
         public void Put(HttpContext context, X509Certificate2 certificate, AuthenticateResult result)
-            => _cache.Set(ComputeKey(certificate), result.Clone(), new MemoryCacheEntryOptions()
-                .SetSize(1).SetSlidingExpiration(_options.CacheEntryExpiration).SetAbsoluteExpiration(certificate.NotAfter));
+        {
+            // Cache expired certs for little while too
+            var absExpiration = (certificate.NotAfter < DateTime.Now) ? DateTime.Now + _options.CacheEntryExpiration : certificate.NotAfter;
```

Comment:
Is `NotAfter` a UTC or local time? Change to `DateTime.UtcNow` if it is UTC, or add a comment that it is local

Reply:
NotAfter is local time, so I'll add a comment, will look to see if its easy to add a unit test